In [100]:
import pandas as pd

In [2]:
df = pd.read_csv('dataset/titanic.csv')

In [29]:
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

In [3]:
df.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

describe() - It gives the statistical detail (mean,count,standard deviation,etc.) about those attributes which are numerical is nature.

In [146]:
df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

From the above table we figure out some points-
* The count of Age column is 714 out of 891 which means we have some missing values.
* The average fare that was paid is 32.204.
* After arranging the dataset in ascending order every 25th index element is given in 25% row. 

In [27]:
df.dtypes[df.dtypes == 'object']        #Checking which attributes has object datatype

Name        object
Sex         object
Ticket      object
Cabin       object
Embarked    object
dtype: object

Knowing all the categorical data using describe() method

In [30]:
df[df.dtypes[df.dtypes == 'object'].index].describe()

Name   Sex  Ticket    Cabin Embarked
count                       891   891     891      204      889
unique                      891     2     681      147        3
top     Braund, Mr. Owen Harris  male  347082  B96 B98        S
freq                          1   577       7        4      644

### Fetching all the numerical and categorical data

In [82]:
num_df = df[df.dtypes[df.dtypes != 'object'].index]

In [86]:
num_lst = []
for i in num_df:
    num_lst.append(i)

print(num_lst)      #All the attribute having numerical value

['PassengerId', 'Survived', 'Pclass', 'Age', 'SibSp', 'Parch', 'Fare']


In [87]:
df[num_lst]         #Printing all the numerical values

PassengerId  Survived  Pclass   Age  SibSp  Parch     Fare
0              1         0       3  22.0      1      0   7.2500
1              2         1       1  38.0      1      0  71.2833
2              3         1       3  26.0      0      0   7.9250
3              4         1       1  35.0      1      0  53.1000
4              5         0       3  35.0      0      0   8.0500
..           ...       ...     ...   ...    ...    ...      ...
886          887         0       2  27.0      0      0  13.0000
887          888         1       1  19.0      0      0  30.0000
888          889         0       3   NaN      1      2  23.4500
889          890         1       1  26.0      0      0  30.0000
890          891         0       3  32.0      0      0   7.7500

[891 rows x 7 columns]

In [51]:
cat_df = df[df.dtypes[df.dtypes == 'object'].index]

In [89]:
cat_lst = []
for i in cat_df:
    cat_lst.append(i)

cat_lst

['Name', 'Sex', 'Ticket', 'Cabin', 'Embarked']

In [90]:
df[cat_lst]

Name     Sex  \
0                              Braund, Mr. Owen Harris    male   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female   
2                               Heikkinen, Miss. Laina  female   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female   
4                             Allen, Mr. William Henry    male   
..                                                 ...     ...   
886                              Montvila, Rev. Juozas    male   
887                       Graham, Miss. Margaret Edith  female   
888           Johnston, Miss. Catherine Helen "Carrie"  female   
889                              Behr, Mr. Karl Howell    male   
890                                Dooley, Mr. Patrick    male   

               Ticket Cabin Embarked  
0           A/5 21171   NaN        S  
1            PC 17599   C85        C  
2    STON/O2. 3101282   NaN        S  
3              113803  C123        S  
4              373450   NaN        S  
..                ...   ...      ...  
886            211536   NaN        S  
887            112053   B42        S  
888        W./C. 6607   NaN        S  
889            111369  C148        C  
890            370376   NaN        Q  

[891 rows x 5 columns]

### Slicing in Dataframe

In [91]:
# If we want to get the all the numeric value from index 10 to 20 we can use slicing similar to list

df[num_lst][10:21]

PassengerId  Survived  Pclass   Age  SibSp  Parch     Fare
10           11         1       3   4.0      1      1  16.7000
11           12         1       1  58.0      0      0  26.5500
12           13         0       3  20.0      0      0   8.0500
13           14         0       3  39.0      1      5  31.2750
14           15         0       3  14.0      0      0   7.8542
15           16         1       2  55.0      0      0  16.0000
16           17         0       3   2.0      4      1  29.1250
17           18         1       2   NaN      0      0  13.0000
18           19         0       3  31.0      1      0  18.0000
19           20         1       3   NaN      0      0   7.2250
20           21         0       2  35.0      0      0  26.0000

In [92]:
# Return the even indexed numerical data

df[num_lst][10:21:2]


PassengerId  Survived  Pclass   Age  SibSp  Parch     Fare
10           11         1       3   4.0      1      1  16.7000
12           13         0       3  20.0      0      0   8.0500
14           15         0       3  14.0      0      0   7.8542
16           17         0       3   2.0      4      1  29.1250
18           19         0       3  31.0      1      0  18.0000
20           21         0       2  35.0      0      0  26.0000

### Adding new column into the dataframe

In [93]:
df['new_col'] = 0       #Inserting a new column with default value as 0

In [94]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  new_col  
0      0         A/5 21171   7.2500   NaN        S        0  
1      0          PC 17599  71.2833   C85        C        0  
2      0  STON/O2. 3101282   7.9250   NaN        S        0  
3      0            113803  53.1000  C123        S        0  
4      0            373450   8.0500   NaN        S        0

### Adding the values of two columns

In [95]:
df['new_col1'] = df['PassengerId'] + df['Pclass']

In [97]:
df.head()

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  new_col  new_col1  
0      0         A/5 21171   7.2500   NaN        S        0         4  
1      0          PC 17599  71.2833   C85        C        0         3  
2      0  STON/O2. 3101282   7.9250   NaN        S        0         6  
3      0            113803  53.1000  C123        S        0         5  
4      0            373450   8.0500   NaN        S        0         8

### Finding the categories of data Present in each column

In [113]:
pd.Categorical(df['Pclass'])        #It return data type and all the categories of data in a column

[3, 1, 3, 1, 3, ..., 2, 1, 3, 1, 3]
Length: 891
Categories (3, int64): [1, 2, 3]

In [114]:
pd.Categorical(df['Survived'])

[0, 1, 1, 1, 0, ..., 0, 1, 0, 1, 0]
Length: 891
Categories (2, int64): [0, 1]

In [115]:
pd.Categorical(df['Cabin'])

[NaN, 'C85', NaN, 'C123', NaN, ..., NaN, 'B42', NaN, 'C148', NaN]
Length: 891
Categories (147, object): ['A10', 'A14', 'A16', 'A19', ..., 'F38', 'F4', 'G6', 'T']

In [117]:
df['Cabin'].unique()        #If we want to see all the unique values

array([nan, 'C85', 'C123', 'E46', 'G6', 'C103', 'D56', 'A6',
       'C23 C25 C27', 'B78', 'D33', 'B30', 'C52', 'B28', 'C83', 'F33',
       'F G73', 'E31', 'A5', 'D10 D12', 'D26', 'C110', 'B58 B60', 'E101',
       'F E69', 'D47', 'B86', 'F2', 'C2', 'E33', 'B19', 'A7', 'C49', 'F4',
       'A32', 'B4', 'B80', 'A31', 'D36', 'D15', 'C93', 'C78', 'D35',
       'C87', 'B77', 'E67', 'B94', 'C125', 'C99', 'C118', 'D7', 'A19',
       'B49', 'D', 'C22 C26', 'C106', 'C65', 'E36', 'C54',
       'B57 B59 B63 B66', 'C7', 'E34', 'C32', 'B18', 'C124', 'C91', 'E40',
       'T', 'C128', 'D37', 'B35', 'E50', 'C82', 'B96 B98', 'E10', 'E44',
       'A34', 'C104', 'C111', 'C92', 'E38', 'D21', 'E12', 'E63', 'A14',
       'B37', 'C30', 'D20', 'B79', 'E25', 'D46', 'B73', 'C95', 'B38',
       'B39', 'B22', 'C86', 'C70', 'A16', 'C101', 'C68', 'A10', 'E68',
       'B41', 'A20', 'D19', 'D50', 'D9', 'A23', 'B50', 'A26', 'D48',
       'E58', 'C126', 'B71', 'B51 B53 B55', 'D49', 'B5', 'B20', 'F G63',
       'C62 C64',

Applying Condition on a particular Column

In [132]:
# Finding all the record in which the age is greater than 18

df[df['Age'] > 18]

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
885          886         0       3   
886          887         0       2   
887          888         1       1   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
885               Rice, Mrs. William (Margaret Norton)  female  39.0      0   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  new_col  new_col1  
0        0         A/5 21171   7.2500   NaN        S        0         4  
1        0          PC 17599  71.2833   C85        C        0         3  
2        0  STON/O2. 3101282   7.9250   NaN        S        0         6  
3        0            113803  53.1000  C123        S        0         5  
4        0            373450   8.0500   NaN        S        0         8  
..     ...               ...      ...   ...      ...      ...       ...  
885      5            382652  29.1250   NaN        Q        0       889  
886      0            211536  13.0000   NaN        S        0       889  
887      0            112053  30.0000   B42        S        0       889  
889      0            111369  30.0000  C148        C        0       891  
890      0            370376   7.7500   NaN        Q        0       894  

[575 rows x 14 columns]

In [134]:
print(f"Number of people whose age is greater than 18: ",len(df[df['Age'] > 18]))
print(f"Number of people whose age is less than 18: ",len(df) - len(df[df['Age'] > 18]))   

Number of people whose age is greater than 18:  575
Number of people whose age is less than 18:  316


In [152]:
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

In [157]:
df1 = df.describe()
print(f"Mean value of age: ",df1['Age']['mean'])

Mean value of age:  29.69911764705882


In [162]:
# Check whose fare was less than the average fair.
df[df['Fare'] < df1['Fare']['mean']]

PassengerId  Survived  Pclass                                      Name  \
0              1         0       3                   Braund, Mr. Owen Harris   
2              3         1       3                    Heikkinen, Miss. Laina   
4              5         0       3                  Allen, Mr. William Henry   
5              6         0       3                          Moran, Mr. James   
7              8         0       3            Palsson, Master. Gosta Leonard   
..           ...       ...     ...                                       ...   
886          887         0       2                     Montvila, Rev. Juozas   
887          888         1       1              Graham, Miss. Margaret Edith   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
889          890         1       1                     Behr, Mr. Karl Howell   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch            Ticket     Fare Cabin Embarked  
0      male  22.0      1      0         A/5 21171   7.2500   NaN        S  
2    female  26.0      0      0  STON/O2. 3101282   7.9250   NaN        S  
4      male  35.0      0      0            373450   8.0500   NaN        S  
5      male   NaN      0      0            330877   8.4583   NaN        Q  
7      male   2.0      3      1            349909  21.0750   NaN        S  
..      ...   ...    ...    ...               ...      ...   ...      ...  
886    male  27.0      0      0            211536  13.0000   NaN        S  
887  female  19.0      0      0            112053  30.0000   B42        S  
888  female   NaN      1      2        W./C. 6607  23.4500   NaN        S  
889    male  26.0      0      0            111369  30.0000  C148        C  
890    male  32.0      0      0            370376   7.7500   NaN        Q  

[680 rows x 12 columns]

In [176]:
df[df['Fare'] == 0][['Name','Pclass']]     #Those ppl who paid no fare

Name  Pclass
179               Leonard, Mr. Lionel       3
263             Harrison, Mr. William       1
271      Tornquist, Mr. William Henry       3
277       Parkes, Mr. Francis "Frank"       2
302   Johnson, Mr. William Cahoone Jr       3
413    Cunningham, Mr. Alfred Fleming       2
466             Campbell, Mr. William       2
481  Frost, Mr. Anthony Wood "Archie"       2
597               Johnson, Mr. Alfred       3
633     Parr, Mr. William Henry Marsh       1
674        Watson, Mr. Ennis Hastings       2
732              Knight, Mr. Robert J       2
806            Andrews, Mr. Thomas Jr       1
815                  Fry, Mr. Richard       1
822   Reuchlin, Jonkheer. John George       1

In [177]:
# How many people has choosen for class 1 ticket?

df[df['Pclass'] == 1]


PassengerId  Survived  Pclass  \
1              2         1       1   
3              4         1       1   
6              7         0       1   
11            12         1       1   
23            24         1       1   
..           ...       ...     ...   
871          872         1       1   
872          873         0       1   
879          880         1       1   
887          888         1       1   
889          890         1       1   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
6                              McCarthy, Mr. Timothy J    male  54.0      0   
11                            Bonnell, Miss. Elizabeth  female  58.0      0   
23                        Sloper, Mr. William Thompson    male  28.0      0   
..                                                 ...     ...   ...    ...   
871   Beckwith, Mrs. Richard Leonard (Sallie Monypeny)  female  47.0      1   
872                           Carlsson, Mr. Frans Olof    male  33.0      0   
879      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
889                              Behr, Mr. Karl Howell    male  26.0      0   

     Parch    Ticket     Fare        Cabin Embarked  
1        0  PC 17599  71.2833          C85        C  
3        0    113803  53.1000         C123        S  
6        0     17463  51.8625          E46        S  
11       0    113783  26.5500         C103        S  
23       0    113788  35.5000           A6        S  
..     ...       ...      ...          ...      ...  
871      1     11751  52.5542          D35        S  
872      0       695   5.0000  B51 B53 B55        S  
879      1     11767  83.1583          C50        C  
887      0    112053  30.0000          B42        S  
889      0    111369  30.0000         C148        C  

[216 rows x 12 columns]

In [179]:
print(f"Number of 1st Class Passenger: ",len(df[df['Pclass'] == 1]))

Number of 1st Class Passenger:  216


In [186]:
# Record who have paid a fair of more than 32 and that person is female

df[(df['Sex'] == 'female') & (df['Fare'] > 32)]


PassengerId  Survived  Pclass  \
1              2         1       1   
3              4         1       1   
31            32         1       1   
43            44         1       2   
52            53         1       1   
..           ...       ...     ...   
853          854         1       1   
856          857         1       1   
863          864         0       3   
871          872         1       1   
879          880         1       1   

                                                  Name     Sex   Age  SibSp  \
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
31      Spencer, Mrs. William Augustus (Marie Eugenie)  female   NaN      1   
43            Laroche, Miss. Simonne Marie Anne Andree  female   3.0      1   
52            Harper, Mrs. Henry Sleeper (Myna Haxtun)  female  49.0      1   
..                                                 ...     ...   ...    ...   
853                          Lines, Miss. Mary Conover  female  16.0      0   
856         Wick, Mrs. George Dennick (Mary Hitchcock)  female  45.0      1   
863                  Sage, Miss. Dorothy Edith "Dolly"  female   NaN      8   
871   Beckwith, Mrs. Richard Leonard (Sallie Monypeny)  female  47.0      1   
879      Potter, Mrs. Thomas Jr (Lily Alexenia Wilson)  female  56.0      0   

     Parch         Ticket      Fare Cabin Embarked  
1        0       PC 17599   71.2833   C85        C  
3        0         113803   53.1000  C123        S  
31       0       PC 17569  146.5208   B78        C  
43       2  SC/Paris 2123   41.5792   NaN        C  
52       0       PC 17572   76.7292   D33        C  
..     ...            ...       ...   ...      ...  
853      1       PC 17592   39.4000   D28        S  
856      1          36928  164.8667   NaN        S  
863      2       CA. 2343   69.5500   NaN        S  
871      1          11751   52.5542   D35        S  
879      1          11767   83.1583   C50        C  

[104 rows x 12 columns]

In [194]:
# Passenger who have paid the maximum Fare

df[df['Fare'] == max(df['Fare'])]

PassengerId  Survived  Pclass                                Name  \
258          259         1       1                    Ward, Miss. Anna   
679          680         1       1  Cardeza, Mr. Thomas Drake Martinez   
737          738         1       1              Lesurer, Mr. Gustave J   

        Sex   Age  SibSp  Parch    Ticket      Fare        Cabin Embarked  
258  female  35.0      0      0  PC 17755  512.3292          NaN        C  
679    male  36.0      0      1  PC 17755  512.3292  B51 B53 B55        C  
737    male  35.0      0      0  PC 17755  512.3292         B101        C

#### iloc() and loc()
* Both method extracts the rows or column from the dataset.
* Difference: -
>* .iloc() (inbuilt/integer) - Based on the inbuilt location it extracts the rows or column. It ignores the upper-bound during selection.
>* loc() (named) - Based on the named indexes it use to extract the data. It considers the upper-bound during selection.

In [195]:
df.iloc[0:2]

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   

   Parch     Ticket     Fare Cabin Embarked  
0      0  A/5 21171   7.2500   NaN        S  
1      0   PC 17599  71.2833   C85        C

In [196]:
df.loc[0:2]

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S

In [199]:
df.iloc[0:2,['PassengerId','Survived','Pclass']]    #it will throw a error because it consider only the inbuilt indexes

IndexError: .iloc requires numeric indexers, got ['PassengerId' 'Survived' 'Pclass']

In [201]:
df.iloc[0:2,[0,1,2,3]]

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   

                                                Name  
0                            Braund, Mr. Owen Harris  
1  Cumings, Mrs. John Bradley (Florence Briggs Th...

In [198]:
df.loc[0:2,['PassengerId','Survived','Pclass']]

PassengerId  Survived  Pclass
0            1         0       3
1            2         1       1
2            3         1       3